spec/v1/signature/jws.go

```diff
@@ -0,0 +1,56 @@
+package signature
+
+const (
+	// MediaTypeJWSEnvelope describes the media type of the JWS envelope.
+	MediaTypeJWSEnvelope = "application/vnd.cncf.notary.v2.jws.v1"
+)
+
+// JWSNotaryClaim is a Notary private claim.
+type JWSNotaryClaim struct {
+	Subject Descriptor `json:"subject"`
+}
+
+// JWSPayload contains the set of claims used by Notary V2.
+type JWSPayload struct {
+	// Private claim.
+	Notary JWSNotaryClaim `json:"notary"`
+
+	// Identifies the number of seconds since Epoch at which the signature was issued.
+	IssuedAt int64 `json:"iat"`
+
+	// Identifies the number of seconds since Epoch at which the signature must not be considered valid.
+	ExpiresAt int64 `json:"exp,omitempty"`
+}
+
+// JWSProtectedHeader contains the set of protected headers.
+type JWSProtectedHeader struct {
+	// Defines which algorithm was used to generate the signature.
+	Algorithm string `json:"alg"`
+
+	// Media type of the secured content (the payload).
+	ContentType string `json:"cty"`
+}
+
+// JWSUnprotectedHeader contains the set of unprotected headers.
+type JWSUnprotectedHeader struct {
+	// RFC3161 time stamp token Base64-encoded.
+	TimeStampToken string `json:"timestamp,omitempty"`
+
+	// List of X.509 certificates, each one Base64-encoded.
+	CertChain []string `json:"x5c"`
```

Comment:
Should the `CertChain` be `[][]byte`? The JSON encoder can take care of the base64 encoding.

Reply:
Wow, I didn't know the json package encodes []byte using `base64`. Then yes, `TimeStampToken` and `CertChain` would better be byte slices.

Reply:
Nit, you can add this link for [x5c](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.6) in the a comment and mention that it's base64 DER.

Reply:
Done!